rclpy/rclpy/client.py

```diff
@@ -71,6 +72,23 @@ def call(self, req):
         self.response = None
         self.sequence_number = _rclpy.rclpy_send_request(self.client_handle, req)
 
+    def service_is_ready(self):
+        return _rclpy.rclpy_service_server_is_available(self.node_handle, self.client_handle)
+
+    def wait_for_service(self, timeout_sec=None):
```

Comment:
TL;DR -1 on this. The current implementation matches what other timeouts use in rclpy to wait forever as well as some python native modules.

`None` is used here to match the other timeout default in rclpy (e.g https://github.com/ros2/rclpy/blob/2ee3c9062fe9ea34ddf535e0230b4cc878fd2d00/rclpy/rclpy/__init__.py#L36)
If we change it we should change it everywhere to stay consistent in the rclpy API.

Regarding the value of the default, given that the rest of the stack is using negative values to wait forever, `-1` would be a more suitable candidate than infinity.
Looking at some python native API: the threading module uses either [None](https://docs.python.org/3/library/threading.html#threading.Event.wait) or [-1](https://docs.python.org/3/library/threading.html#threading.Lock.acquire) to represent wait forever.

Reply:
> If we change it we should change it everywhere to stay consistent in the rclpy API.

Yeah, after looking at other parts of rclpy, I see that we use `None` for infinity; we should remain consistent here.  That being said, I still find `None` unsatisfying, since we really want to specify wait forever (which is infinity).  For now, my suggestion would be to keep `None` in the API, but then use the infinity trick in the implementation (which still gets rid of the special cases of `None` in the loop).  But I definitely won't block merging it for that, so I'll leave it to @sloretz whether he wants to change the implementation.

Reply:
@clalancette `float('inf')` used in implementation b5ef8f0a7eade2ec3277956b606a03a0e637f3dd